lang/scala.md

```diff
@@ -0,0 +1,66 @@
+---
+lang: Scala
+syntax: scala
+title: Scala is a modern programming language designed and created by Martin Odersky.
+author: techysubham
+---
+
+```scala
+object HelloWorld {
+   def main(args: Array[String]) {
+      println("Hello, World!")
+   }
+} 
+```
+
+## What is Scala?
+
+[Scala](https://en.wikipedia.org/wiki/Scala_(programming_language)) is a modern programming language designed and created by [Martin Odersky](https://en.wikipedia.org/wiki/Martin_Odersky). The design of the language started in 2001 and was released to the public in early 2004. Martin Odersky had a huge hand in the implementation of javac (the primary Java compiler) and also designed Generic Java, a facility of generic programming that was added to the Java programming language in 2004. This is why it doesn’t come as a surprise that Scala is similar to Java in many aspects, it’s actually written to run in [JVM (Java Virtual Machine)](https://en.wikipedia.org/wiki/Java_virtual_machine).
+
+## Why to use Scala?
+
+- Scala has an exact syntax, eliminating boilerplate code. Programs written in Scala require less code than similar programs written in Java.
+- It is both an object-oriented language and a functional language. This combination makes Scala the right choice for web development.
+- Scala uses an expressive typing system that ensures statistical abstraction is secure and consistent.
+- It’s easy to learn, especially for programmers with an object-oriented background with Java or similar language.
+- Scala is highly scalable, useful for building fault-tolerant, highly concurrent systems
+- It’s ideal for data analytics when supported by tools like Apache Spark
+
+
+## Setting up Scala development environment
+
+- Step 1: Verifying Java Packages
+- The first thing we need to have is a Java Software Development Kit(SDK) installed on the computer. We need to verify these SDK packages and if not installed then install them. Open the command window and type in the following commands:
```

Comment:
Provide a Wikipedia link to the Java SDK. Check other instances where it's needed as well.